Hunting Queries/SigninLogs/UserAccounts-NewSingleFactorAuth.yaml

```diff
@@ -30,6 +36,18 @@ query: |
     | where AuthenticationDetails != "[]"
     | extend AuthenticationMethod = tostring(todynamic(AuthenticationDetails)[0].authenticationMethod)
     | where AuthenticationMethod != "Previously satisfied"
+    | join kind=leftouter (
+        BehaviorAnalytics

```

Comment:
You call the same query blocks 2 here:
      BehaviorAnalytics
      | where ActivityType in ("FailedLogOn", "LogOn")
      | where isnotempty(SourceIPAddress)
      | project UsersInsights, DevicesInsights, ActivityInsights, InvestigationPriority, SourceIPAddress
      | project-rename IPCustomEntity = SourceIPAddress
      | summarize
          UsersInsights = make_set(UsersInsights, 1000),
          DevicesInsights = make_set(DevicesInsights, 1000),
          IPInvestigationPriority = sum(InvestigationPriority)
      by IPCustomEntity

It would be more efficient to materialize that query element at the start of the query and then just refer to the materialized values - https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/materializefunction

Reply:
Makes sense, changed this.